### PR TITLE
Feature/64/add review point manage page

### PR DESCRIPTION
**Pull request by syoung125**
리뷰에디터 리스트, 리뷰포인트 상세 내역 페이지 추가

### CODE REVIEW

**Comment by pickk-dev** [comment]
#### 📝Changed routes:


| [**/manage/question/\[id\]**](https://pickk-super-admin-pmdhdsvy3.vercel.app/manage/question/[id]) | [**/manage/user/\[id\]**](https://pickk-super-admin-pmdhdsvy3.vercel.app/manage/user/[id]) |
|:-:|:-:|
|<a href="https://pickk-super-admin-pmdhdsvy3.vercel.app/manage/question/[id]">Screenshot of /manage/question/[id]</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/6c_pW0fuQDLsm_NuIbic3FVjtPI_.png">(view full size)</a> |<a href="https://pickk-super-admin-pmdhdsvy3.vercel.app/manage/user/[id]">Screenshot of /manage/user/[id]</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/YWifq9ONwi3E_Tgdd5yITwDP9Gk_.png">(view full size)</a> |



| [**/manage/user**](https://pickk-super-admin-pmdhdsvy3.vercel.app/manage/user) |
|:-:|
|<a href="https://pickk-super-admin-pmdhdsvy3.vercel.app/manage/user">Screenshot of /manage/user</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/qFvLGu7l2iiO89e5f8BrVSokhx0_.png">(view full size)</a> |







Commit b894328f1a439ed75cf030465b2bf31b6da4e024 (https://pickk-super-admin-pmdhdsvy3.vercel.app).

File: src/components/user/review-point-detail/modal.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        setDetailReason(value);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleSignClick = (value) => {
+    setSign(value);
+  };
```
Comment (greatSumini) [inline src/components/user/review-point-detail/modal.tsx:43]:
```suggestion
  const toggleSign = (value: SignType) => () => {
    handleFormChange({
      target: {
        name: 'sign',
        value,
      }
    })
  };
```

File: src/components/user/review-point-detail/modal.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        sign,
+        amount: point,
+        ...(reason.length && { reason }),
+        ...(detailReason.length && { detailReason }),
+      });
+      router.reload();
+    }
+
+    isSetModalVisible(false);
+    initField();
```
Comment (greatSumini) [inline src/components/user/review-point-detail/modal.tsx:60]: 1. escape case에서 return 추가해주세요!
2. await function에 try-catch 추가해주세요

File: src/components/user/review-point-detail/modal.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  };
+
+  const handleCancel = () => {
+    isSetModalVisible(false);
+    initField();
```
Comment (greatSumini) [inline src/components/user/review-point-detail/modal.tsx:65]: 이 부분이 반복돼서 함수로 빼면 좋을 것 같아요~

File: src/components/user/review-point-detail/modal.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+import styled from 'styled-components';
+import { Input, Modal } from 'antd';
+import TextArea from 'antd/lib/input/TextArea';
+
+import { Button, P } from '@src/component/atoms';
+import { INDIGO_BLUE, FIRE_RED } from '@src/component/atoms/colors';
+
+import { SignType } from '@src/types';
+import ReviewPointService from '@src/services/ReviewPoint';
+
```
Comment (greatSumini) [inline src/components/user/review-point-detail/modal.tsx:12]: const FORM_INITIAL_VALUE = {
  amount: 0,
  sign: null,
  reason: '',
  detailReason: '',
}

File: src/components/user/review-point-detail/modal.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    isSetModalVisible(false);
+    initField();
+  };
+
+  const initField = () => {
+    setPoint(0);
+    setSign(null);
+    setReason('');
+    setDetailReason('');
+  };
```
Comment (greatSumini) [inline src/components/user/review-point-detail/modal.tsx:73]:
```suggestion
  const initForm = () => {
    setForm(FORM_INITIAL_VALUE)
  };
```

File: src/components/user/review-point-detail/modal.tsx [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+          sign={SignType.Plus}
+          selected={sign === SignType.Plus}
+          onClick={() => handleSignClick(SignType.Plus)}
+        />
+        <StyledButton
+          title="차감"
+          sign={SignType.Minus}
+          selected={sign === SignType.Minus}
+          onClick={() => handleSignClick(SignType.Minus)}
+        />
```
Comment (greatSumini) [inline src/components/user/review-point-detail/modal.tsx:103]:
```suggestion
        <StyledButton
          title="적립"
          sign={SignType.Plus}
          selected={sign === SignType.Plus}
          onClick={toggleSign(SignType.Plus)}
        />
        <StyledButton
          title="차감"
          sign={SignType.Minus}
          selected={sign === SignType.Minus}
          onClick={toggleSign(SignType.Minus)}
        />
```

1. 이중 화살표 함수를 사용해서 리팩토링할 수 있습니다.
2. map 사용을 시도해보세요~

File: src/interfaces/User/IReviewPoint.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  lookId: number;
+  sign: SignType;
+  amount: number;
+  reason?: string;
+  detailReason?: string;
+  createdAt: string;
+  updatedAt: string;
+}
+
+export type ReviewPointInputDTO = {
```
Comment (greatSumini) [inline src/interfaces/User/IReviewPoint.ts:15]: Pick을 사용해보세요!

File: src/types/User.ts [hunk 8]
```diff
@@ -41,3 +41,8 @@ export enum UploaderType {
   Editor = 'EDITOR',
   None = 'NONE',
 }
+
+export enum SignType {
```
Comment (greatSumini) [inline src/types/User.ts:45]: PointSign을 사용해도 될 것 같아요!

File: src/components/user/list/filter.tsx [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+import BoardSectionWrapper from '@src/component/templates/border-section-wrapper';
+import Select from '@src/component/organisms/select';
+import { P, Space } from '@src/component/atoms';
+
+import { UploaderType } from '@src/types';
+import { getUploaderTypeIconComponent } from '@src/lib/User';
+
+const { Option } = _Select;
+
+function UserListFilter({ filter, setFilter }) {
```
[the remark this answers is not in the document]
Comment (syoung125) [reply src/components/user/list/filter.tsx]: a6e9c13

File: src/components/user/list/filter.tsx [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+          ),
+        ]}
+        onChange={handleChange}
+        value={filter.uploaderType || '모두'}
+      />
+      <Space direction="ROW" size={20} />
+      <Row>
+        <P>닉네임: </P>
+        <Space direction="ROW" size={8} />
+        <Search
```
[the remark this answers is not in the document]
Comment (syoung125) [reply src/components/user/list/filter.tsx:102]: 3a0bc72

File: src/components/user/review-point-detail/modal.tsx [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+      case 'reason':
+        setReason(value);
+        break;
+      case 'detailReason':
+        setDetailReason(value);
+        break;
+      default:
+        break;
+    }
+  };
```
[the remark this answers is not in the document]
Comment (syoung125) [reply src/components/user/review-point-detail/modal.tsx]: 4c02938

File: src/interfaces/User/IReviewPoint.ts (hunk 7 above)
Comment (syoung125) [reply src/interfaces/User/IReviewPoint.ts]: d7b788f

File: src/types/User.ts (hunk 8 above)
Comment (syoung125) [reply src/types/User.ts]: 88ac113

**Pull request merged**